sysrib/server_test.go

```diff
@@ -156,6 +156,173 @@ func configureInterface(t *testing.T, intf *AddIntfAction, yclient *ygnmi.Client
 	}
 }
 
+var (
+	// v4v6Mapper maps IPv4 addresses to IPv6 addresses for testing v6.
+	v4v6Mapper = map[string]string{
+		"192.168.1.1/24": "2001::1/49",
+		"192.168.2.1/24": "2002::1/49",
+		"192.168.3.1/24": "2003::1/49",
+		"192.168.4.1/24": "2004::1/49",
+		"192.168.5.1/24": "2005::1/49",
+		"192.168.1.0/24": "2001::/49",
+		"192.168.2.0/24": "2002::/49",
+		"192.168.3.0/24": "2003::/49",
+		"192.168.4.0/24": "2004::/49",
+		"192.168.5.0/24": "2005::/49",
+		"192.168.1.42":   "2001::42",
+		"192.168.2.42":   "2002::42",
+		"192.168.3.42":   "2003::42",
+		"192.168.4.42":   "2004::42",
+		"192.168.5.42":   "2005::42",
+		"10.0.0.0/8":     "1000::/40",
+		"20.0.0.0/8":     "2100::/40",
+		"30.0.0.0/8":     "3000::/40",
+		"40.0.0.0/8":     "4000::/40",
+		"10.0.0.0":       "1000::",
+		"20.0.0.0":       "2100::",
+		"30.0.0.0":       "3000::",
+		"40.0.0.0":       "4000::",
+		"10.10.10.10":    "1000::10",
+		"20.10.10.10":    "2100::10",
+		"15.0.0.0":       "1500::",
+		"11.10.10.10":    "1100::10",
+		"192.0.2.1/30":   "2001:1::1/49",
+		"192.0.2.5/30":   "2001:2::1/49",
+		"192.0.2.9/30":   "2001:3::1/49",
+		"192.0.2.0/30":   "2001:1::/49",
+		"192.0.2.4/30":   "2001:2::/49",
+		"192.0.2.8/30":   "2001:3::/49",
+		// GUE only
+		"10.10.20.20": "1000:0:20::20",
+	}
+
+	v4v6PolicyMap = map[GUEHeaders]GUEHeaders{
+		{
+			GUEPolicy: GUEPolicy{
+				dstPortv4: 8,
+				srcIP4:    [4]byte{42, 42, 42, 42},
+				isV6:      false,
+			},
+			dstIP4: [4]byte{192, 168, 1, 42},
+		}: {
+			GUEPolicy: GUEPolicy{
+				dstPortv6: 8,
+				srcIP6:    [16]byte{42, 42, 42, 42, 42},
+				isV6:      true,
+			},
+			dstIP6: [16]byte{0x20, 0x1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0x42},
```

Comment:
maybe a test utility to go from ip str to array might be nice

Reply:
There is the tradeoff of [16]byte forcing a very precise definition, so it's not yet clear that such a utility will be better. Maybe with more use cases we will see a pattern.